elements/lisk-p2p/src/peer_selection.ts

```diff
@@ -25,11 +32,11 @@ interface HistogramValues {
 }
 
 /* tslint:enable: readonly-keyword */
-export const selectPeers = (
-	peers: ReadonlyArray<P2PDiscoveredPeerInfo>,
-	nodeInfo?: P2PNodeInfo,
-	numOfPeers: number = 0,
+export const selectPeersForRequest = (
```

Comment:
Shouldn't this request function should always return one peer? we can avoid unnecessary conditions to check the value of `peerLimit` and should only default to `1`?

Reply:
unless we want to support in the future to request multiple peers at a time?

Reply:
Now yes. But in the future (or for a sidechain) we could request from multiple peers and calculate some kind of consensus on the responses. This is based on a discussion I had a while back with @diego-G